src/libpriv/rpmostree-scripts.c

```diff
@@ -957,6 +967,34 @@ rpmostree_deployment_sanitycheck (int           rootfs_fd,
   rpmostree_bwrap_append_child_argv (bwrap, "/usr/bin/true", NULL);
   if (!rpmostree_bwrap_run (bwrap, cancellable, error))
     return FALSE;
+
   sd_journal_print (LOG_INFO, "sanitycheck(/usr/bin/true) successful");
   return TRUE;
 }
+
+/* Check that we can load the rpmdb. See
+ * https://github.com/projectatomic/rpm-ostree/issues/1566.
+ *
+ * This is split out of the one above for practical reasons: the check above runs right
+ * after scripts are executed to give a nicer error if the scripts did `rm -rf`.
+ */
+gboolean
+rpmostree_deployment_sanitycheck_rpmdb (int           rootfs_fd,
+                                        GCancellable *cancellable,
+                                        GError      **error)
+{
+  g_autoptr(RpmOstreeRefSack) sack = rpmostree_get_refsack_for_root (rootfs_fd, ".", error);
+  if (!sack)
+    return FALSE;
+
+  /* And check we can query pkgs and that they show up as installed. */
+  const char *sanity_pkgs[] = {"glibc", "filesystem", NULL};
```

Comment:
I still hesitate on this - this is a fairly strong hardcoding, I'm worried it's going to come back to bite us.

In theory, people could use rpm-ostree with non-Fedora distributions - do those have `filesystem`?
And what if we decided to remove `filesystem` in the future?  (The whole `filesystem`/`setup` situation is actually batshit insane)

Particularly when thinking about *rebases* - we could be looking at an arbitrary tree from the future.

Reply:
How about: If there are layered packages or overrides, check for those packages?  If there aren't any modifications to the tree, then the rpmdb (if any) will have been written by libostree, not librpm - and hence subject to the consistent rigorous errno checks we do pulling there.

Reply:
OK, I updated this! We now only check the rpmdb in the local assembly cases, and we check for specific packages if there are any, otherwise just checking that there are *some* packages in the rpmdb. :arrow_down: 

This got a little more complex than I was envisioning. Though I do like the idea of checking that the packages we just installed indeed show up as installed.